### PR TITLE
First step towards the simulation challenge

### PR DESCRIPTION
**Pull request by victor-gonzalez**
For the time being is being a separate task
The final goal is a single one

Implemented
- data collection directly in differential form
- huge decrease in memory layout

### CODE REVIEW

**Comment by victor-gonzalez** [comment]
None of the four failed checks seem to have anything related with what I have done

**Comment by victor-gonzalez** [comment]
After today's WP4+WP14 meeting I've realized that the PR is big and perhaps I should have explained it a bit better.

The first commit basically replicates `dptpdtcorrelations.cxx` with two major changes in the `DptDptCorrelationsTask` which only involve variable definitions and their corresponding manipulations in the `init` and `process` methods.

The singles variables are indexed by track `one` and `two` and the pair variables are indexed by the four potential combinations. Few service functions were templated preparing the MC support. The main change is, apart that we simultaneously collect the four track combinations, that we do it directly in `delta eta`, `delta phi`. The gain in the memory and results file layouts is huge. This parallel the changes already introduced in the `AliPhysics` framework so that we will compare apples to apples when we run the benchmark.

The second commit just extracted the data collection out of the `init` and `process` methods encapsulating it into the `DataCollectingEngine` structure.

The third commit just removed the span of a task per centrality range and used instead a `DataCollectingEngine` per centrality range.

The fourth commit just changed the name of the task and of the source file to comply with the executable naming convention

The fifth commit just modified a sentence to comply with o2 code format

File: Analysis/Tasks/PWGCF/dptdptcorrelations-simchl.cxx [hunk 1]
```diff
@@ -0,0 +1,927 @@
...
+    TProfile* fhSum2PtPtnw_vsC[4];   //!<! un-weighted accumulated \f${p_T}_1 {p_T}_2\f$ distribution vs event centrality 1-1,1-2,2-1,2-2, combinations
+    TProfile* fhSum2DptDptnw_vsC[4]; //!<! un-weighted accumulated \f$\sum ({p_T}_1- <{p_T}_1>) ({p_T}_2 - <{p_T}_2>) \f$ distribution vs \f$\Delta\eta,\;\Delta\phi\f$ distribution vs event centrality 1-1,1-2,2-1,2-2, combinations
+
+    const char* tname[2] = {"1", "2"}; ///< the external track names, one and two, for histogram creation
+    const char* trackPairsNames[4] = {"OO", "OT", "TO", "TT"};
+
+    /// \brief Returns the potentially phi origin shifted phi
+    /// \param phi the track azimuthal angle
+    /// \return the track phi origin shifted azimuthal angle
+    inline float GetShiftedPhi(float phi)
```
Comment (iarsene) [inline Analysis/Tasks/PWGCF/dptdptcorrelations-simchl.cxx]: Is the "inline" needed here? I thought that the functions defined inside the structure are inlined by default

File: Analysis/Tasks/PWGCF/dptdptcorrelations-simchl.cxx [hunk 2]
```diff
@@ -0,0 +1,927 @@
...
+    /// According to the bining structure, to the track eta will correspond
+    /// a zero based bin index and similarlly for the track phi
+    /// The returned index is the composition of both considering eta as
+    /// the first index component
+    /// WARNING: for performance reasons no checks are done about the consistency
+    /// of track's eta and phin with the corresponding ranges so, it is suppossed
+    /// the track has been accepted and it is within that ranges
+    /// IF THAT IS NOT THE CASE THE ROUTINE WILL PRODUCE NONSENSE RESULTS
+    template <typename TrackObject>
+    inline int GetEtaPhiIndex(TrackObject const& t)
```
Comment (iarsene) [inline Analysis/Tasks/PWGCF/dptdptcorrelations-simchl.cxx]: Same comment about the inline here. If they are not needed, i suggest to remove them. It makes the code easier and more readable.

**Comment by iarsene** [comment]
Hi @victor-gonzalez,
The PR is in general fine with me. I just had a comment regarding the inline keywords. 
Cheers,
Ionut

File: Analysis/Tasks/PWGCF/dptdptcorrelations-simchl.cxx (hunk 1 above)
Comment (victor-gonzalez) [reply Analysis/Tasks/PWGCF/dptdptcorrelations-simchl.cxx]: According to my knowledge, only explicit `inline` functions are expanded inline, but I might be wrong

File: Analysis/Tasks/PWGCF/dptdptcorrelations-simchl.cxx (hunk 2 above)
Comment (victor-gonzalez) [reply Analysis/Tasks/PWGCF/dptdptcorrelations-simchl.cxx]: You were right! https://isocpp.org/wiki/faq/inline-functions
I will update the PR accordingly

**Comment by victor-gonzalez** [comment]
You are right with the `inline` functions (https://isocpp.org/wiki/faq/inline-functions)
I will update the PR accordingly

**Comment by victor-gonzalez** [comment]
```
/sw/SOURCES/O2/6530-slc8_x86-64/0//Analysis/Tasks/PWGCF/dptdptcorrelations-simchl.cxx:1:1: error: missing or malformed copyright notice
```
Where can I find the right copyright notice?
The other errors seem to not be related with what I did

**Comment by victor-gonzalez** [comment]
@iarsene @jgrosseo  any stopper for this PR?

**Comment by iarsene** [comment]
Hi Victor,
It is fine with me, but i don't know about the new "cs8" tests which are pending. I see they are pending for other PRs as well.
I will find out and see how to proceed.
Ionut

**Comment by victor-gonzalez** [comment]
@iarsene @jgrosseo @ktf  `macOS`, `o2` and `o2-cs8` CI tests keep failing with reason not related to the changes I made

**Comment by jgrosseo** [comment]
Merging so that you can go ahead with your work.
What about using the histogram registry?
There is a typo in filteranalyistask

**Pull request merged**